ndcube/ndcube_sequence.py

```diff
@@ -239,6 +241,46 @@ def explode_along_axis(self, axis):
         # creating a new sequence with the result_cubes keeping the meta and common axis as axis
         return self._new_instance(result_cubes, common_axis=new_common_axis, meta=self.meta)
 
+    def combine_cubes(self, axis=0, common_wcs_index=0):
+        """
+        Reprojects all `~ndcube.NDCube` objects to a common WCS, and stacks the data together
+        to return a single (N+1)-dimensional `~ndcube.NDCube` with an associated Compound WCS
+        containing extra axes corresponding to the sequence axis.
+
+        Parameters
+        ----------
+        axis: `int`, optional
+            The axis along which the data is to be stacked.
+
+        common_wcs_index: `int`, optional
+            The index of the `~ndcube.NDCube` whose WCS is to be used as the base on which to
+            reproject all other `~ndcube.NDCube`s in the `~ndcube.NDCubeSequence`.
+
+        Returns
+        -------
+        `ndcube.NDCube`
+            A combined (N+1)D cube with stacked data of all cubes.
+        """
+        from ndcube import NDCube  # adding this here to prevent a circular import
+
+        # Reproject all cubes to a common WCS
+        target_wcs = self[common_wcs_index].wcs
+        shape_out = self[common_wcs_index].data.shape
+        reprojected_cubes = [cube.reproject(target_wcs, shape_out=shape_out,
+                                            return_footprint=False) for cube in self]
+
+        # Stack data of all cubes together
+        combined_data = np.stack([cube.data for cube in reprojected_cubes], axis=axis)
+
+        sequence_axis = list(self.sequence_axis_coords.values())[0]
+        sequence_axis_wcs = QuantityTableCoordinate(u.Quantity(sequence_axis)).wcs
+
+        # Create a Compound WCS object
+        compound_wcs = CompoundLowLevelWCS(target_wcs, sequence_axis_wcs,
+                                           mapping=[i for i in range(target_wcs.pixel_n_dim + 1)])
```

Comment:
I would expect that you would need to involve `axis` in this computation. If you haven't put the new pixel axis first then the new WCS axis doesn't want to be last?

Reply:
Yeah, this computation needs to be improved. It would also depend on whether we want to drop the axis parameter entirely as you mentioned in the other comment.